src/Microsoft.Health.Fhir.Core/Features/Operations/Reindex/ReindexProcessingJob.cs

```diff
@@ -174,18 +164,10 @@ private async Task ProcessQueryAsync(CancellationToken cancellationToken)
 
                 if (!cancellationToken.IsCancellationRequested)
                 {
-                    try
+                    if (result != null)
```

Comment:
Is result being null an expected case? ProcessSearchResultsAsync will throw an exception if result is null, so this check will never be false.